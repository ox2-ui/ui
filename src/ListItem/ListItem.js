import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import Thumb from '@ox2/thumb/Thumb';

/**
 * ListItem Component
 */
const ListItem = props => {
  const {
    className: classNameProp,
    title,
    subtitle1,
    subtitle2,
    image,
    pictureBgColor,
    icon,
    iconGroup,
    iconColor,
    pictureWidth,
    pictureHeight,
    onClick,
    _id,
    hidePicture,
    style,
    widget,
    widgetBottom,
  } = props;

  const className = classNames(
    'border:bottom border:2 border-color:neutral-faded',
    classNameProp,
  );

  return (
    <div className={className} style={style}>
      <div
        className="font:roboto-condensed display:flex flex:items-center color:white padding:10 select:none cursor:pointer"
        onClick={() => onClick(_id)}
      >
        {!hidePicture &&
          <Thumb
            color={pictureBgColor}
            height={pictureHeight}
            icon={icon}
            iconColor={iconColor}
            iconGroup={iconGroup}
            image={image}
            width={pictureWidth}
          />}
        {/* class min-w:0 added to fix Flex child truncate issue */}
        <div className="display:flex flex:column flex:content-center margin-l:10 min-w:0 line-h:130p">
          {title &&
            <div className="select:text size:18 line-h:120p text:truncate text-color:brand">
              {title}
            </div>}
          {subtitle1 &&
            <div className="select:text size:16 line-h:115p text:truncate text-color:neutral-dim">
              {subtitle1}
            </div>}
          {subtitle2 &&
            <div className="select:text size:16 line-h:115p text:truncate text-color:neutral-dim">
              {subtitle2}
            </div>}
        </div>
        {widget &&
          <div className="margin-l:auto flex:0-auto">
            {widget(props)}
          </div>}
      </div>
      {widgetBottom &&
        <div>
          {widgetBottom(props)}
        </div>}
    </div>
  );
};

ListItem.defaultProps = {
  className: '',
  hidePicture: false,
  icon: '',
  iconColor: '',
  iconGroup: '',
  image: '',
  pictureBgColor: '',
  pictureHeight: 60,
  pictureWidth: 60,
  style: {},
  subtitle1: '',
  subtitle2: '',
  title: '',
  widget: () => null,
  widgetBottom: () => null,
};

ListItem.propTypes = {
  /**
   * The css class name of the root element.
   */
  _id: PropTypes.string.isRequired,
  className: PropTypes.string,
  hidePicture: PropTypes.bool,
  icon: PropTypes.string,
  iconColor: PropTypes.string,
  iconGroup: PropTypes.string,
  image: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  pictureBgColor: PropTypes.string,
  pictureHeight: PropTypes.number,
  pictureWidth: PropTypes.number,
  style: PropTypes.object,
  subtitle1: PropTypes.string,
  subtitle2: PropTypes.string,
  title: PropTypes.string,
  widget: PropTypes.func,
  widgetBottom: PropTypes.func,
};

export default ListItem;
