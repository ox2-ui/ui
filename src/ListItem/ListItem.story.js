import React from 'react';
import { storiesOf } from '@kadira/storybook';
import ListItem from './ListItem';
import Icon from '@ox2/icon/Icon';
import Button from '@ox2/button/Button';

const handler = () => console.log('🍇'); // eslint-disable-line no-console

const handlerButtons = () => console.log('💋'); // eslint-disable-line no-console

const widget = () => {
  return <Icon icon={'server'} size={'20px'} className={'text-color:critical margin:7'} />;
};

const widgetBottom = () => {
  return (
    <div className="display:flex flex:content-end padding-x:10 padding-b:7">
      <Button onClick={handlerButtons} btn="tiny outline neutral-dim" className="margin-r:7"> Duplicate </Button>
      <Button onClick={handlerButtons} btn="tiny outline neutral-dim"> Preview </Button>
    </div>
  );
};

storiesOf('ListItem', module)
  .add('regular', () => (
    <ListItem
      _id={'1nNKyP6h6pMMjAyTMK'}
      title={'Jane Sanders'}
      subtitle1={'Ox2 Limited'}
      subtitle2={'Marketing Manager'}
      icon={'user'}
      onClick={handler}
    />
  ))
  .add('one subtitle', () => (
    <ListItem
      _id={'1nNKyP6h6pMMjAyTMK'}
      title={'Jane Sanders'}
      subtitle1={'Ox2 Limited'}
      icon={'user'}
      onClick={handler}
    />
  ))
  .add('no subtitle', () => (
    <ListItem
      _id={'1nNKyP6h6pMMjAyTMK'}
      title={'Jane Sanders'}
      icon={'user'}
      onClick={handler}
    />
  ))
  .add('colored icon', () => (
    <ListItem
      _id={'1nNKyP6h6pMMjAyTMK'}
      title={'Jane Sanders'}
      subtitle1={'Ox2 Limited'}
      subtitle2={'Marketing Manager'}
      icon={'user'}
      pictureBgColor={'white'}
      iconColor={'neutral-light'}
      onClick={handler}
    />
  ))
  .add('image', () => (
    <ListItem
      _id={'2nNKyP6h6pMMjAyTMK'}
      title={'Jane Sanders'}
      subtitle1={'Ox2 Limited'}
      subtitle2={'Marketing Manager'}
      image={'https://randomuser.me/api/portraits/women/21.jpg'}
      icon={'user'}
      onClick={handler}
    />
  ))
  .add('widget', () => (
    <ListItem
      _id={'2nNKyP6h6pMMjAyTMK'}
      title={'Jane Sanders'}
      subtitle1={'Ox2 Limited'}
      subtitle2={'Marketing Manager'}
      image={'https://randomuser.me/api/portraits/women/21.jpg'}
      icon={'user'}
      onClick={handler}
      widget={widget}
    />
  ))
  .add('bottom widget', () => (
    <ListItem
      _id={'2nNKyP6h6pMMjAyTMK'}
      title={'Jane Sanders'}
      subtitle1={'Ox2 Limited'}
      subtitle2={'Marketing Manager'}
      image={'https://randomuser.me/api/portraits/women/21.jpg'}
      icon={'user'}
      onClick={handler}
      widgetBottom={widgetBottom}
    />
  ))
  .add('both widgets', () => (
    <ListItem
      _id={'2nNKyP6h6pMMjAyTMK'}
      title={'Jane Sanders'}
      subtitle1={'Ox2 Limited'}
      subtitle2={'Marketing Manager'}
      image={'https://randomuser.me/api/portraits/women/21.jpg'}
      icon={'user'}
      onClick={handler}
      widget={widget}
      widgetBottom={widgetBottom}
    />
  ));
