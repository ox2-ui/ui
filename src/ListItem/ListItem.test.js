/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ListItem from './ListItem';

const handler = () => console.log('🍇'); // eslint-disable-line no-console


it('renders correctly', () => {
  const wrapper = mount(
    <ListItem
      _id={'1nNKyP6h6pMMjAyTMK'}
      title={'Jane Sanders'}
      subtitle1={'Ox2 Limited'}
      subtitle2={'Marketing Manager'}
      icon={'user'}
      onClick={handler}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
