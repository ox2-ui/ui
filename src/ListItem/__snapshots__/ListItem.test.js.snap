exports[`test renders correctly 1`] = `
<ListItem
  _id="1nNKyP6h6pMMjAyTMK"
  icon="user"
  onClick={[Function]}
  subtitle1="Ox2 Limited"
  subtitle2="Marketing Manager"
  title="Jane Sanders">
  <div
    className="border:bottom border:2 border-color:neutral-faded">
    <div
      className="font:roboto-condensed display:flex flex:items-center color:white padding:10 select:none cursor:pointer"
      onClick={[Function]}>
      <Thumb
        height={60}
        icon="user"
        width={60}>
        <div
          className="display:flex flex:items-center flex:content-center border:rounded color:neutral"
          style={
            Object {
              "WebkitFlex": "0 0 60px",
              "flex": "0 0 60px",
              "height": "60px",
              "width": "60px",
            }
          }>
          <Icon
            className="text-color:white"
            icon="user"
            size="42px">
            <span
              className="icon-user text-color:white"
              style={
                Object {
                  "color": undefined,
                  "fontSize": "42px",
                }
              } />
          </Icon>
        </div>
      </Thumb>
      <div
        className="display:flex flex:column flex:content-center margin-l:10 min-w:0 line-h:130p">
        <div
          className="select:text size:18 line-h:120p text:truncate text-color:brand">
          Jane Sanders
        </div>
        <div
          className="select:text size:16 line-h:115p text:truncate text-color:neutral-dim">
          Ox2 Limited
        </div>
        <div
          className="select:text size:16 line-h:115p text:truncate text-color:neutral-dim">
          Marketing Manager
        </div>
      </div>
    </div>
  </div>
</ListItem>
`;
